internal/rows/arrowbased/batchloader.go

```diff
@@ -27,246 +27,281 @@ type BatchIterator interface {
 	Close()
 }
 
-type BatchLoader interface {
-	rowscanner.Delimiter
-	GetBatchFor(recordNum int64) (SparkArrowBatch, dbsqlerr.DBError)
-	Close()
-}
+func NewCloudBatchIterator(
```

Comment:
can `BatchLoader` be used by the client of the driver

Reply:
@jackyhu-db No. Everything in `internal/` folder is private to the library, user cannot access those files